File: sfml demos/exercise 04/SFML/World.cpp
#include "World.h"
#include "Game.h"
#include <math.h>

World::World()
{
	Vector2f origin = Vector2f(0.f, 0.f);

	//sky preenche 1/3 da screen
	sky = *new RectangleShape(Vector2f(Game::screenWidth, Game::screenHeight / 3.f));
	sky.setOrigin(origin);
	sky.setFillColor(Color::Blue);
	sky.setPosition(0, 0);

	street = *new RectangleShape(Vector2f(Game::screenWidth, Game::screenHeight / 2.f));
	street.setFillColor(*new Color(127, 127, 127));
	street.setOrigin(origin);
	street.setPosition(0, 200);
	
	//calcula a hipotenusa correspondente a metade da tela horizontal (ponto de fuga da estrada) com a altura até o sky (1/3 screen)
	float hipo = sqrt((Game::screenWidth / 2 * Game::screenWidth / 2) + (2 * Game::screenHeight / 3 * 2 * Game::screenHeight / 3));

	lineStreetLeft = *new RectangleShape(Vector2f(hipo, 5));
	lineStreetLeft.setFillColor(Color::White);
	lineStreetLeft.setOrigin(origin);
	lineStreetLeft.setPosition(0, Game::screenHeight);
	lineStreetLeft.rotate(-45);

	lineStreetRight = *new RectangleShape(Vector2f(hipo, 5));
	lineStreetRight.setFillColor(Color::White);
	lineStreetRight.setOrigin(origin);
	lineStreetRight.setPosition(Game::screenWidth, Game::screenHeight);
	lineStreetRight.rotate(-135);

	forestLeft.setPointCount(3);
	forestLeft.setFillColor(Color::Green);
	forestLeft.setPoint(0, Vector2f(Game::screenWidth / 2, Game::screenHeight / 3));
	forestLeft.setPoint(1, Vector2f(0, Game::screenHeight / 3));
	forestLeft.setPoint(2, Vector2f(0, Game::screenHeight));

	forestRight.setPointCount(3);
	forestRight.setFillColor(Color::Green);
	forestRight.setPoint(0, Vector2f(Game::screenWidth / 2, Game::screenHeight / 3));
	forestRight.setPoint(1, Vector2f(Game::screenWidth, Game::screenHeight / 3));
	forestRight.setPoint(2, Vector2f(Game::screenWidth, Game::screenHeight));
}


World::~World()
{
}

File: C#/exercise 03/sfml.net/src/World.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SFML.Graphics;
using SFML.System;

namespace sfml.net.src
{
    class World
    {
        #region Fields

        private RectangleShape sky;
        private RectangleShape lineStreetLeft;
        private RectangleShape lineStreetRight;
        private RectangleShape street;

        private ConvexShape forestLeft;
        private ConvexShape forestRight;

        #endregion


        #region Public

        public World()
        {
            var origin = new Vector2f(0.0f, 0.0f);

            //sky preenche 1/3 da screen
            sky = new RectangleShape(new Vector2f(Game.WINDOW_WIDTH, Game.WINDOW_HEIGHT / 3));
            sky.Origin = origin;
            sky.FillColor = Color.Blue;
            sky.Position = new Vector2f(0, 0);

            street = new RectangleShape(new Vector2f(Game.WINDOW_WIDTH, Game.WINDOW_HEIGHT));
            street.FillColor = (new Color(127, 127, 127));
            street.Origin = origin;
            street.Position = new Vector2f(0, 200);

            //calcula a hipotenusa correspondente a metade da tela horizontal (ponto de fuga da estrada) com a altura até o sky (1/3 screen)
            var hipo = Math.Sqrt((double)((Game.WINDOW_WIDTH / 2 * Game.WINDOW_WIDTH / 2) + (2 * Game.WINDOW_HEIGHT / 3 * 2 * Game.WINDOW_HEIGHT / 3)));

            lineStreetLeft = new RectangleShape(new Vector2f((float)hipo, 5));
            lineStreetLeft.FillColor = Color.White;
            lineStreetLeft.Origin = origin;
            lineStreetLeft.Position = new Vector2f(0, Game.WINDOW_HEIGHT);
            lineStreetLeft.Rotation = -45;

            lineStreetRight = new RectangleShape(new Vector2f((float)hipo, 5));
            lineStreetRight.FillColor = Color.White;
            lineStreetRight.Origin = origin;
            lineStreetRight.Position = new Vector2f(Game.WINDOW_WIDTH, Game.WINDOW_HEIGHT);
            lineStreetRight.Rotation = -135;

            forestLeft = new ConvexShape(3);
            forestLeft.FillColor = Color.Green;
            forestLeft.SetPoint(0, new Vector2f(Game.WINDOW_WIDTH / 2, Game.WINDOW_HEIGHT / 3));
            forestLeft.SetPoint(1, new Vector2f(0, Game.WINDOW_HEIGHT / 3));
            forestLeft.SetPoint(2, new Vector2f(0, Game.WINDOW_HEIGHT));

            forestRight = new ConvexShape(3);
            forestRight.FillColor = Color.Green;
            forestRight.SetPoint(0, new Vector2f(Game.WINDOW_WIDTH / 2, Game.WINDOW_HEIGHT / 3));
            forestRight.SetPoint(1, new Vector2f(Game.WINDOW_WIDTH, Game.WINDOW_HEIGHT / 3));
            forestRight.SetPoint(2, new Vector2f(Game.WINDOW_WIDTH, Game.WINDOW_HEIGHT));
        }

        public void Display(RenderTarget window)
        {
            window.Draw(sky);
            window.Draw(street);
            window.Draw(forestLeft);
            window.Draw(forestRight);
            window.Draw(lineStreetLeft);
            window.Draw(lineStreetRight);
        }

        #endregion
    }
}
